### DISCUSSION TITLE
`aws configure` uses USERPROFILE instead of HOME

### DISCUSSION THREAD

**Discussion by rcdailey** [aws/aws-cli#7035] [Q&A] [answered]
### Describe the bug

When I have the `HOME` environment variable explicitly set on my Windows 10 system, running `aws configure` from a Powershell terminal results in the `.aws` directory and its files being created at `%USERPROFILE%/.aws` instead of `%HOME%/.aws`.

### Expected Behavior

If the `HOME` environment variable is set, the `.aws` directory should be placed at `%HOME%/.aws` instead of `%USERPROFILE%/.aws`, according to [the docs](https://docs.aws.amazon.com/sdkref/latest/guide/file-location.html).

### Current Behavior

The `aws configure` command will spit out files at `%USERPROFILE%/.aws` but the .NET library correctly looks in `%HOME%/.aws` which doesn't exist because the tool did not place the files in the right location.

### Reproduction Steps

1. In Windows Environment Variables, add a new **user** environment variable named `HOME` and set it to some path like `C:\linux_home`.
2. Restart machine.
3. Execute `aws configure` and follow the prompts. Insert a value for each question.
4. Observe that files are created in `%USERPROFILE%/.aws` which is incorrect.

### AWS .NET SDK and/or Package version used

```xml
<PackageReference Include="AWSSDK.Core" Version="3.7.8.10" />
<PackageReference Include="AWSSDK.EC2" Version="3.7.61" />
<PackageReference Include="AWSSDK.Extensions.NETCore.Setup" Version="3.7.2" />
<PackageReference Include="AWSSDK.SecretsManager" Version="3.7.2.18" />
```

### Targeted .NET Platform

net6.0

### Operating System and version

Windows 10

**Comment by ashishdhingra** [comment]
@rcdailey Thanks for reporting the issue. The issue related to `aws configure` should be opened in https://github.com/aws/aws-cli repository.

**Comment by tim-finnigan** [comment]
Hi @rcdailey thanks for reaching out. In the [documentation](https://docs.aws.amazon.com/sdkref/latest/guide/file-location.html) that you linked to, it does mention how Windows uses `%USERPROFILE%`. And in the .NET developer guide it describes how the SDK looks for your credentials there: https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/creds-assign.html

So this is the expected behavior. Since this isn’t a bug, I’m going to convert the issue into a Discussion per our guidance process. Please let us know if you have any follow up questions.

**Comment by rcdailey** [comment, marked as answer]
@tim-finnigan First of all, I think this should have been left as an issue until both sides agree. Furthermore setting your own response as the answer is also not that helpful IMHO. I'm not upset or anything, I just feel like the way my bug report is being handled is rather dismissive.

More on topic: Something is absolutely broken, because you have a tool that writes to one location and an API that reads from another. The implementations are not consistent. At this point, I'm asserting the library is behaving correctly and the `aws` tool is broken.

Please check my link one more time. Specifically note that it says the following (emphasis is mine):

> A ~/ or ~ followed by the file system's default path separator at the start of the path is resolved by checking, **in order**,
>
> 1. (All platforms) The HOME environment variable
> 2. (Windows platforms) The USERPROFILE environment variable

Note that number 1 should happen first, that is, that `HOME` is checked first. This is supported on *all platforms*. Thus, `USERPROFILE` should only be used if `HOME` is not defined. This is why I assert that the behavior of the library is correct.

Please reconsider moving this back to an issue and wait for both sides to agree on the conclusion before hastily migrating things to discussions.

Thank you for your quick response!

**Reply by tim-finnigan** [reply]
Hi @rcdailey apologies for moving hastily on this one. We can certainly convert this back to an issue.

But regarding your original point that running `aws configure` in Windows uses `%USERPROFILE%` is documented and expected behavior. It is referenced [here](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-where) in the CLI documentation as well.

So maybe we should consider this to be a feature request for adding consistency with the HOME and USERPROFILE environment variables across platforms. What do you think?

**Reply by rcdailey** [reply]
I don't feel like this is a feature request. I'm not saying that it is unexpected, only that it is out of order. Remember point 1 from the quoted text I provided earlier? `HOME` should be used first, *then* `USERPROFILE`. And even if I'm wrong, we can't deny or ignore the inconsistent behavior between the `aws` tool and the C# library.

I must not be wording this properly as I feel like I'm repeating myself at this point. I apologize that I'm not able to get my point across.

**Reply by tim-finnigan** [reply]
Hi @rcdailey, I created a new issue to try and capture everything we've discussed here: https://github.com/aws/aws-cli/issues/7051

I think there are a few things that require further investigation, and I hope that the new issue provides a helpful framing for more research into this. Please feel free to add any more feedback or insights on this topic. Apologies again for any misunderstandings on my end.
